### PR TITLE
unverifed chains status

### PR DESCRIPTION
**Pull request by 0xbala-k**
Closes https://github.com/allinbits/demeris-backend/issues/224

### CODE REVIEW

**Comment by 0xbala-k** [comment]
The failing test will pass after https://github.com/allinbits/emeris-utils/pull/35 is merged.

File: api/database/errors.go [hunk 1]
```diff
@@ -2,12 +2,11 @@ package database
 
 import "fmt"
 
-type ErrNoMatchingChannel struct {
+type ErrNoDestChain struct {
```
Comment (sahith-narahari) [inline api/database/errors.go:5]: Any reason we're renaming this, old one seems more appropriate

File: api/database/ibc_channel.go [hunk 2]
```diff
@@ -7,68 +7,117 @@ import (
...
-	SELECT
-		c1.chain_name AS chain_a_chain_name,
-		c1.channel_id AS chain_a_channel_id,
-		c1.counter_channel_id AS chain_a_counter_channel_id,
-		c1.chain_id AS chain_a_chain_id,
-		c2.chain_name AS chain_b_chain_name,
-		c2.channel_id AS chain_b_channel_id,
-		c2.counter_channel_id AS chain_b_counter_channel_id,
-		c2.chain_id AS chain_b_chain_id
+	// q := `
```
Comment (sahith-narahari) [inline api/database/ibc_channel.go]: please remove commented code, or add comments why it would be needed

File: api/router/correlation_test.go [hunk 3]
```diff
@@ -51,7 +51,7 @@ func TestCorrelationIDMiddleWare(t *testing.T) {
 				count++
 			}
 		}
-		return count == 2
+		return count == 4
```
Comment (sahith-narahari) [inline api/router/correlation_test.go]: can you explain why this is changing in this pr

File: api/chains/chains.go [hunk 4]
```diff
@@ -503,55 +503,73 @@ func VerifyTrace(c *gin.Context) {
...
+					"err",
+					err,
+				)
+			}
+			return
+		}
+
+		if primaryChannelInfo.ChannelName != channel {
+
+			cause := fmt.Sprintf("%s : not primary channel for chain %s- expecting %s got %s", hash, chainName, primaryChannelInfo.ChannelName, channel)
```
Comment (sahith-narahari) [inline api/chains/chains.go]: please break it into two lines forbetter readability```suggestion
			cause := fmt.Sprintf("%s : not primary channel for chain %s- expecting %s got %s",
				hash, chainName, primaryChannelInfo.ChannelName, channel)
```

File: api/chains/migration_test.go [hunk 5]
```diff
@@ -0,0 +1,144 @@
...
+// Creates tracelistner database and required tables only if they dont exist
+func runTraceListnerMigrations(t *testing.T) {
+	for _, m := range migrations {
+		_, err := testingCtx.CnsDB.Instance.DB.Exec(m)
+		require.NoError(t, err)
+	}
+}
+
+// Empties all tracelistner tables
+func truncateTracelistener(t *testing.T) {
```
Comment (sahith-narahari) [inline api/chains/migration_test.go]: are we testing migrations or is this a test for the updated db query?

**Review by sahith-narahari — CHANGES_REQUESTED** [review]

File: api/chains/migration_test.go (hunk 5 above)
Comment (0xbala-k) [reply api/chains/migration_test.go]: Just to clear all the tables so that we don't get data from previous test case.

Comment (sahith-narahari) [reply api/chains/migration_test.go]: let me rephrase it, what are we testing in this file. If it's db queries, don't we have a base setup to do that?

File: api/router/correlation_test.go (hunk 3 above)
Comment (0xbala-k) [reply api/router/correlation_test.go]: It was just a small change depending on the log request PR in utils. So thought do it in this PR instead of creating another.

File: api/database/errors.go (hunk 1 above)
Comment (0xbala-k) [reply api/database/errors.go:5]: Cause we use this error with `GetIbcChannelToChain` function which tries to find the destination chain info when source chain name,id and channel id is passed.

File: api/chains/migration_test.go (hunk 5 above)
Comment (0xbala-k) [reply api/chains/migration_test.go]: Okay, Just moved these migrations to utils. Please have a look.

**Review by sgerogia — APPROVED** [review]

File: api/chains/chains.go [hunk 6]
```diff
@@ -475,7 +475,7 @@ func VerifyTrace(c *gin.Context) {
 			} else {
 				e1 := deps.NewError(
 					"denom/verify-trace",
-					fmt.Errorf("failed querying for %s", hash),
+					fmt.Errorf("failed querying for %s, error:%w", hash, err),
```
Comment (DeshErBojhaa) [inline api/chains/chains.go]: Space b2in `error:%w` -> `error: %w`?

**Review by DeshErBojhaa — APPROVED** [review]
Please addr the comment I left.

**Review by PrathyushaLakkireddy — APPROVED** [review]

**Review by akhilkumarpilli — APPROVED** [review]

**Review by sahith-narahari — APPROVED** [review]
